ballerina-tests/http-advanced-tests/tests/relaxed_data_binding_negative_tests.bal

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+import ballerina/http;
+import ballerina/http_test_common as common;
+
+type Employee record {
+    int id;
+    string name;
+    string address?;
+};
+
+type Department record {
+    string name;
+    int? employeeCount;
+};
+
+type Lead record {|
+    int id;
+    string name;
+|};
+
+listener http:Listener laxDataBindingNegativeEP = new (laxDataBindingNegativeTestPort, httpVersion = http:HTTP_2_0);
+final http:Client laxNegativeClient = check new ("http://localhost:" + laxDataBindingNegativeTestPort.toString(), {
+    httpVersion: http:HTTP_2_0
+});
+
+service /company on laxDataBindingNegativeEP {
+
+    resource function post test1(@http:Payload Employee employee) returns Employee {
+        return employee;
+    }
+
+    resource function post test2(@http:Payload Department department) returns Department {
+        return department;
+    }
+
+    resource function post test3(@http:Payload Lead lead) returns Lead {
+        return lead;
+    }
+
+    resource function get test4() returns json {
+        return {
+            "id": 1001,
+            "name": "Tharmigan",
+            "address": null
+        };
+    }
+
+    resource function get test5() returns json {
+        return {
+            "name": "Engineering"
+        };
+    }
+
+    resource function get test6() returns json {
+        return {
+            "id": 1001,
+            "name": "Danesh",
+            "address": "Piliyandala",
+            "Experience": 10
+        };
+    }
+}
+
+@test:Config
+function testNullForOptionalFieldNegative() returns error? {
+    Employee|error response = laxNegativeClient->/company/test1.post({ "id": 1001, "name": "Tharmigan", "address": null});
+    if (response is http:ClientRequestError) {
```

Comment:
No need for parentheses here:
```suggestion
    if response is http:ClientRequestError {
```

Lets try to follow the best practices: https://learn-ballerina.github.io/best_practices/language_best_practices.html

Check other places as well